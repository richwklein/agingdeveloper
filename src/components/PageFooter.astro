---
/**
 * @name PageFooter
 * The component used to render the page footer.

 * @param {Props} props - The properties of the page footer.
 * @return {Function} an Astro component factory function.
 *
 * @example
 * <PageFooter />
 */
import LinkExternal from "@components/LinkExternal.astro"
import { Icon } from "astro-icon/components"
import type { CollectionEntry } from "astro:content"

interface Props {
  site: CollectionEntry<"site">
  class?: string
}

const { site: entry, class: className } = Astro.props
const site = entry.data
const year = new Date().getFullYear()
---

<footer class:list={["container mx-auto grid-rows-3 p-4 xl:max-w-screen-xl", className]}>
  <div>
    <ol class:list={["flex items-center gap-1"]}>
      <li>
        <LinkExternal
          to="/rss.xml"
          title={`${site.title} - RSS`}
          class:list={["block rounded-full p-2", "hover:bg-gray-100 focus:bg-gray-300"]}
        >
          <Icon
            name="mdi:rss-feed"
            title={`${site.title} - RSS`}
            class:list={["h-6 w-6 text-primary-main"]}
          />
        </LinkExternal>
      </li>
      <li>
        <LinkExternal
          to={site.repository}
          title={`${site.title} - Repository`}
          class:list={["block rounded-full p-2", "hover:bg-gray-100 focus:bg-gray-300"]}
        >
          <Icon
            name="mdi:github"
            title={`${site.title} - Repository`}
            class:list={["h-6 w-6 text-primary-main"]}
          />
        </LinkExternal>
      </li>
    </ol>
  </div>
  <div class:list={["text-center text-xs"]}>&copy; {year} {site.title}</div>
  <div></div>
</footer>
