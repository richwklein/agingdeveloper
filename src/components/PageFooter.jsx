import {Box, Grid, Typography} from "@mui/material";
import React from "react";
import PropTypes from "prop-types";
import ExternalLink from "./ExternalLink";

/**
 * React component used to render the copyright of the site.
 *
 * @param {CopyrightProps} props - The copyright props.
 * @return {React.ReactElement} - The react component
 *
 * @example
 * <Copyright title={title} repository={repository} />
 *
 * @ignore
 */
const Copyright = ({title, repository}) => {
  return (
    <Box textAlign="center">
      <Typography variant="caption" align="center">
        &copy;
        {" "}
        {new Date().getFullYear()}
        {" "}
        <ExternalLink to={repository}>{title}</ExternalLink>
      </Typography>
    </Box>
  );
};

/**
 * @typedef CopyrightProps - The copyright props.
 * @property {string} title - The title of the site.
 * @property {string} repository - The repository the site code is in.
 */
Copyright.propTypes = {
  title: PropTypes.string.isRequired,
  repository: PropTypes.string.isRequired,
};

/**
 * React component used to render the powered by line.
 *
 * @return {React.ReactElement}
 *
 * @example
 * <PoweredBy />
 *
 * @ignore
 */
const PoweredBy = () => {
  return (
    <Box textAlign="center">
      <Typography variant="caption">
        {"Powered by "}
        <ExternalLink to="https://www.gatsbyjs.org/">Gatsby</ExternalLink>
        {" "}
        and
        {" "}
        <ExternalLink to="https://app.netlify.com">Netlify</ExternalLink>
        .
      </Typography>
    </Box>
  );
};

/**
 * React component used to render the page footer.
 *
 * @param {PageFooterProps} props - The page footer props.
 * @return {React.ReactElement}
 *
 * @example
 * <PageFooter title={title} repository={repository} />
 *
 * @todo RSS icon & maybe scroll to top
 */
const PageFooter = ({title, repository}) => {
  return (
    <Grid
      container
      component="footer"
      sx={{p: 2, mt: 4}}
    >
      <Grid item sm={4} xs={12}/>
      <Grid item sm={4} xs={12}>
        <Copyright title={title} repository={repository} />
        <PoweredBy />
      </Grid>
      <Grid item sm={4} xs={12}/>
    </Grid>
  );
};

/**
 * @typedef PageFooterProps - The page footer props.
 * @property {string} title - The title of the site.
 * @property {string} repository - The repository the site code is in.
 */
PageFooter.propTypes = {
  title: PropTypes.string.isRequired,
  repository: PropTypes.string.isRequired,
};

export default PageFooter;
