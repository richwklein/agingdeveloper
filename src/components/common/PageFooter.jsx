import {Box, Grid, Typography} from "@mui/material";
import React from "react";
import PropTypes from "prop-types";
import ExternalLink from "./ExternalLink";

const Copyright = ({title, repository}) => {
  return (
    <Box textAlign="center">
      <Typography variant="caption" align="center">
        &copy;
        {" "}
        {new Date().getFullYear()}
        {" "}
        <ExternalLink to={repository}>{title}</ExternalLink>
      </Typography>
    </Box>
  );
};

Copyright.propTypes = {
  title: PropTypes.string.isRequired,
  repository: PropTypes.string.isRequired,
};

const PoweredBy = () => {
  return (
    <Box textAlign="center">
      <Typography variant="caption">
        {"Powered by "}
        <ExternalLink to="https://www.gatsbyjs.org/">Gatsby</ExternalLink>
        {" "}
        and
        {" "}
        <ExternalLink to="https://app.netlify.com">Netlify</ExternalLink>
        .
      </Typography>
    </Box>
  );
};

const PageFooter = ({title, repository}) => {
  return (
    <Grid
      container
      component="footer"
      sx={{
        padding: 2,
        marginTop: 4,
      }}
    >
      <Grid item sm={12} md={4} />
      <Grid item sm={12} md={4}>
        <Copyright title={title} repository={repository} />
        <PoweredBy />
      </Grid>
      <Grid item sm={12} md={4} />
    </Grid>
  );
};

PageFooter.propTypes = {
  title: PropTypes.string.isRequired,
  repository: PropTypes.string.isRequired,
};

export default PageFooter;
