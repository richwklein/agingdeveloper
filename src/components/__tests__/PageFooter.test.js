import React from "react";
import {cleanup, render} from "@testing-library/react";
import PageFooter from "../PageFooter";

describe("PageFooter", () => {
  beforeAll(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date("2020-05-01"));
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  afterEach(cleanup);

  const renderComponent = () => {
    return render( <PageFooter title="test-title" repository="https://github.com/test-repository" />);
  };

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that copyright is the correct year.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toContainHTML("Â© 2020");
  });
});
