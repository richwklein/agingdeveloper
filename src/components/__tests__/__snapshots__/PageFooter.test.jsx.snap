// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFooter that it renders correctly. 1`] = `
<footer
  class="MuiGrid-root MuiGrid-container css-bw06ur-MuiGrid-root"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
  />
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-xi606m"
    >
      <span
        class="MuiTypography-root MuiTypography-caption MuiTypography-alignCenter css-1v2k9m3-MuiTypography-root"
      >
        Â©
         
        2020
         
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
          data-link="external"
          href="https://github.com/test-repository"
          rel="noopener noreferrer"
          target="_blank"
        >
          test-title
        </a>
      </span>
    </div>
    <div
      class="MuiBox-root css-xi606m"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        Powered by 
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
          data-link="external"
          href="https://www.gatsbyjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Gatsby
        </a>
         
        and
         
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
          data-link="external"
          href="https://app.netlify.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          Netlify
        </a>
        .
      </span>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
  />
</footer>
`;
